components/automatic-points-table/commons/automatic-points-table.lua

```diff
@@ -0,0 +1,57 @@
+---
+-- @Liquipedia
+-- wiki=commons
+-- page=Module:AutomaticPointsTable
+--
+-- Please see https://github.com/Liquipedia/Lua-Modules to contribute
+--
+
+local Arguments = require('Module:Arguments')
+local Class = require('Module:Class')
+local Json = require('Module:Json')
+local String = require('Module:StringUtils')
+local Table = require('Module:Table')
+
+local AutomaticPointsTable = Class.new(
+	function(self, frame)
+		self.frame = frame
+		self.args = Arguments.getArgs(frame)
+	end
+)
+
+function AutomaticPointsTable.run(frame)
+	local pointsTable = AutomaticPointsTable(frame)
+
+	pointsTable:extractPositionBackgroundData()

```

Comment:
The use of "extract" here implies extraction of the position background data from `pointsTable`, but that's not what you're doing. It would make more sense to make this a part of the constructor, and instead use the verb `parse`, which is often used for processing input from a different source. An alternative, which might be cleaner considering my below comment, could be to create `pointTable:parseInput()`

Reply:
Noted, however parseInput() would be a very long function, I'm going to keep both **parsePoisitionBackgroundData** and **parseTournaments**; will also add **parseTeams** in the next commit.

Reply:
`parseInput` can call the other two :)